src/Sulu/Bundle/AdminBundle/Resources/js/containers/FieldBlocks/FieldBlocks.js

```diff
@@ -13,6 +13,46 @@ const MISSING_BLOCK_ERROR_MESSAGE = 'The "block" field type needs at least one t
 const BLOCK_PREVIEW_TAG = 'sulu.block_preview';
 
 export default class FieldBlocks extends React.Component<FieldTypeProps<Array<BlockEntry>>> {
+    componentDidUpdate(prevProps: FieldTypeProps<Array<BlockEntry>>) {
+        const {defaultType, types, value: oldValues, onChange} = this.props;
+        const {types: oldTypes} = prevProps;
+
+        if (!oldValues) {
+            return;
+        }
+
+        if (!types || !oldTypes) {
+            throw new Error(MISSING_BLOCK_ERROR_MESSAGE);
+        }
+
+        // do nothing when types is same reference as oldTypes
+        if (types === oldTypes) {
+            return;
+        }
+
+        const newValues = toJS(oldValues);
+        const usedTypes = newValues.map((block) => block.type);
```

Comment:
Distinguishing between `oldValues` and `newValues` is a little bit confusing here IMO. At this point both of them are exactly the same, right? I would rename the `oldValues` to just `value` (then it is clearer what they are) and get the `usedTypes` from the `value`. Then you could do the below checks exactly the same, and change the value by calling `onChange` somehow like this:

```javascript
onChange(value.map((blockValue) => ({
    ...blockValue,
    type: types[blockValue.type] ? types[blockValue.type].type : defaultType,
})));
```

That woud work around the thing that is IMO a naming issue.